tests/test_custom_checkers/test_one_iteration_checker.py

```diff
@@ -10,47 +10,65 @@
 class TestOneIterationChecker(pylint.testutils.CheckerTestCase):
     CHECKER_CLASS = OneIterationChecker
 
-    def test_while_with_return(self):
-        src = """
-        def func(x: int) -> int:
-            while x > 5:
-                x -= 1
-                return x
-            return 0
-        """
-        mod = astroid.parse(src)
-        mod.accept(CFGVisitor())
-        while_node = next(mod.nodes_of_class(nodes.While))
-
-        with self.assertAddsMessages(
-            pylint.testutils.MessageTest(
-                msg_id="one-iteration",
-                node=while_node,
+    import pytest  # make sure pytest is imported if not already
+
+    @pytest.mark.parametrize(
+        "src,node_type",
+        [
+            (
+                """
+                def func(x: int) -> int:
+                    while x > 5:
+                        x -= 1
+                        return x
+                    return 0
+                """,
+                nodes.While,
             ),
-            ignore_position=True,
-        ):
-            self.checker.visit_while(while_node)
-
-    def test_while_with_break(self):
-        src = """
-        x = 1
-        while x < 5: #@
-            x += 1
-            break
-        print(x)
-        """
+            (
+                """
+                x = 1
+                while x < 5:
+                    x += 1
+                    break
+                print(x)
+                """,
+                nodes.While,
+            ),
+            (
+                """
+                def func(x: list[int]) -> int:
+                    for i in x:
+                        return i
+                """,
+                nodes.For,
+            ),
+            (
+                """
+                for i in range(10):
+                    break
+                print(i)
+                """,
+                nodes.For,
+            ),
+        ],
+    )
+    def test_one_iteration_simple(self, src, node_type):
```

Comment:
Please call this `test_one_iteration_message` and include all test cases that cause this checker to add a message (e.g., `test_while_with_return_all_branches` and others below). This will go even further with the removal of the code duplication!